### ISSUE TITLE
Issue with one use limited coupon

### ISSUE THREAD

**Issue by diogoceribelli** [magento/magento2#31798]
Magento 2.3.5-p1

### Steps to reproduce (*)
1. Administrator creates a new discount coupon, limited to one use;
2. Two or more users insert products into their shopping cart and apply the discount using the coupon;
3. Two or more users click on checkout at the same time.

### Expected result (*)
1. Even if pressing at the same time, some order must have been placed first;
2. This order will receive the coupon discount, subsequent orders will receive some type of error saying that the coupon has already been used;
3. Now the coupon that had only one use, can no longer be used.

### Actual result (*)
1. All users who pressed the checkout button at the same time will receive the coupon discount even with the one use limit;
2. Only after simultaneous orders will the coupon be deactivated.

The problem then is that purchases made simultaneously will disregard the limit defined in the discount coupon rule. This should not happen, only one of those purchases should receive the discount.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._

**Comment by ajijshekh123** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Bravo** [comment]
Hello @diogoceribelli 

Thank you for your report

This is an already known and Confirmed issue. Please follow all the updates at https://github.com/magento/magento2/issues/21952

We have to close this issue as Duplicate of #21952
